File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0:
            return 0

        half = x

        # while half * half > x :
        #     half = half // 2
        #     print(half, half * half)
        
        l = 1
        h = x

        # print(l,h,x)

        while l <= h :
            half = l + ((h - l) // 2)

            if half == x // half:
                return half

            elif half > x // half :
                h = half -1

            else:
                l = half + 1
            
        
        
        return h

File: 0069-sqrtx/0069-sqrtx.js
/**
 * @param {number} x
 * @return {number}
 */

 // s 2:7
var mySqrt = function(x) {

    let result = 0;

    while (true) {
        if (result * result === x) {
            return result
        }

         if (result * result > x) {
            return result - 1
        }

        result += 1;
    }
};